src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -679,7 +683,18 @@ trait Applications extends Compatibility { self: Typer with Dynamic =>
           }
       case _ =>
     }
-    assignType(cpy.TypeApply(tree)(typedFn, typedArgs), typedFn, typedArgs)
+    if (typedFn.tpe eq TryDynamicCallType) {
+      (pt, typedFn) match {
```

Comment:
Same here: Move the block to a separate function (so that each alternative of
the `if` is a single expression.